test/controllers/utils/ReferenceRetrievalSpec.scala

```diff
@@ -130,69 +148,174 @@ class ReferenceRetrievalSpec extends PlaySpec with Matchers with MockSubscriptio
 
           status(result) mustBe OK
         }
-        "the reference is not in session and we call out to retrieve the reference successfully and add the reference to the session" in {
-          mockRetrieveReferenceSuccess(utr)(reference)
+        "the reference is not in session and we retrieve the reference successfully" when {
+          "there is an existing reference" should {
+            "return OK, do the signupRetrieve auditing and add the reference to the session" in {
 
-          implicit val request: Request[AnyContent] = FakeRequest().withSession(ITSASessionKeys.UTR -> utr)
+              val auditModel: SignupRetrieveAuditModel = SignupRetrieveAuditModel(userType = "individual", None, utr, None)
+              mockRetrieveReference(utr)(Right(Existing(reference)))
 
-          val result = TestReferenceRetrieval.withReference(utr) { reference =>
-            Future.successful(Ok(reference))
+              implicit val request: Request[AnyContent] = FakeRequest().withSession(ITSASessionKeys.UTR -> utr)
+
+              val result = Await.result(TestReferenceRetrieval.withReference(utr) { reference =>
+                Future.successful(Ok(reference))
+              }, Duration.apply("5 secs"))
```

Comment:
```suggestion
              val result = TestReferenceRetrieval.withReference(utr) { reference =>
                Future.successful(Ok(reference))
              }.futureValue
```
You should be able to use either .futureValue like above or use the default play test helper `await(...)` instead of implementing an Await yourself.

Same for below tests

Reply:
fixed